generators/entity/templates/client/angular/src/test/javascript/e2e/entities/_entity.spec.ts

```diff
@@ -25,56 +26,205 @@ if (enableTranslation) {
+    let <%= entityInstance %>ComponentsPage: <%= entityClass %>ComponentsPage;
 
     beforeAll(() => {
         browser.get('/');
-
-        accountMenu.click();
-        login.click();
-
-        username.sendKeys('admin');
-        password.sendKeys('admin');
-        element(by.css('button[type=submit]')).click();
-        browser.waitForAngular();
+        navBarPage.getSignInPage().autoSignInUsing('admin', 'admin');
     });
 
     it('should load <%= entityClassPlural %>', () => {
-        entityMenu.click();
-        element.all(by.css('[routerLink="<%= entityStateName %>"]')).first().click().then(() => {
-            <%_ if (enableTranslation) { _%>
-            const expectVal = /<%= angularAppName %>.<%= entityTranslationKey %>.home.title/;
-            <%_ } else { _%>
-            const expectVal = /<%= entityClassPluralHumanized %>/;
-            <%_ } _%>
-            element.all(by.css('h2 span')).first().<%- elementGetter %>.then((value) => {
-                expect(value).toMatch(expectVal);
-            });
-        });
+        navBarPage.goToEntity('<%= entityStateName %>');
+        <%= entityInstance %>ComponentsPage = new <%= entityClass %>ComponentsPage();
+        <%_ if (enableTranslation) { _%>
+        expect(<%= entityInstance %>ComponentsPage.getTitle()).toMatch(/<%= angularAppName %>.<%= entityTranslationKey %>.home.title/);
+        <%_ } else { _%>
+        expect(<%= entityInstance %>ComponentsPage.getTitle()).toMatch(/<%= entityClassPluralHumanized %>/);
+        <%_ } _%>
     });
 
     it('should load create <%= entityClass %> dialog', () => {
-        element(by.css('button.create-<%= entityUrl %>')).click().then(() => {
-            <%_ if (enableTranslation) { _%>
-            const expectVal = /<%= angularAppName %>.<%= entityTranslationKey %>.home.createOrEditLabel/;
-            <%_ } else { _%>
-            const expectVal = /Create or edit a <%= entityClassHumanized %>/;
-            <%_ } _%>
-            element.all(by.css('h4.modal-title')).first().<%- elementGetter %>.then((value) => {
-                expect(value).toMatch(expectVal);
-            });
-
-            element(by.css('button.close')).click();
+        <%= entityInstance %>ComponentsPage.clickOnCreateButton();
+        <%= entityInstance %>DialogPage = new <%= entityClass %>DialogPage();
+        <%_ if (enableTranslation) { _%>
+        expect(<%= entityInstance %>DialogPage.getModalTitle()).toMatch(/<%= angularAppName %>.<%= entityTranslationKey %>.home.createOrEditLabel/);
+        <%_ } else { _%>
+        expect(<%= entityInstance %>DialogPage.getModalTitle()).toMatch(/Create or edit a <%= entityClassHumanized %>/);
+        <%_ } _%>
+        <%= entityInstance %>DialogPage.close();
+    });
+
+    it('should create and save <%= entityClassPlural %>', () => {
+        <%= entityInstance %>ComponentsPage.clickOnCreateButton();
+        <%_ for (idx in fields) {
```

Comment:
you could do a forEach loop here as it would be less cryptic. I know we still have this in our code here and there but this is from the pre es6 code, we should use es6 atleast on the new code like this